pkg/ddevapp/commands.go

```diff
@@ -1,47 +1,83 @@
 package ddevapp
 
 import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/ddev/ddev/pkg/dockerutil"
 	"github.com/ddev/ddev/pkg/fileutil"
 	"github.com/ddev/ddev/pkg/globalconfig"
 	"github.com/ddev/ddev/pkg/util"
-	copy2 "github.com/otiai10/copy"
-	"os"
-	"path/filepath"
 )
 
 // PopulateCustomCommandFiles sets up the custom command files in the project
 // directories where they need to go.
 func PopulateCustomCommandFiles(app *DdevApp) error {
+	// Remove legacy directory
+	legacyCommandDir := app.GetConfigPath(".global_commands")
+	if fileutil.IsDirectory(legacyCommandDir) {
+		err := fileutil.PurgeDirectory(legacyCommandDir)
+		if err != nil {
+			util.Warning("Unable to remove files from legacy directory %s: %v", legacyCommandDir, err)
+			return nil
+		}
+		err = os.Remove(legacyCommandDir)
+		if err != nil {
+			util.Warning("Unable to remove legacy directory %s: %v", legacyCommandDir, err)
+		}
+	}
 
 	sourceGlobalCommandPath := filepath.Join(globalconfig.GetGlobalDdevDir(), "commands")
 	err := os.MkdirAll(sourceGlobalCommandPath, 0755)
 	if err != nil {
 		return nil
 	}
 
-	projectCommandPath := app.GetConfigPath("commands")
-	// Make sure our target global command directory is empty
-	copiedGlobalCommandPath := app.GetConfigPath(".global_commands")
-	err = os.MkdirAll(copiedGlobalCommandPath, 0755)
+	commandDirs, err := fileutil.ListFilesInDirFullPath(sourceGlobalCommandPath)
 	if err != nil {
-		util.Error("Unable to create directory %s: %v", copiedGlobalCommandPath, err)
-		return nil
+		return err
 	}
 
-	// Make sure it's empty
-	err = fileutil.PurgeDirectory(copiedGlobalCommandPath)
-	if err != nil {
-		util.Error("Unable to remove %s: %v", copiedGlobalCommandPath, err)
-		return nil
-	}
+	for _, serviceDirOnHost := range commandDirs {
```

Comment:
This could also be all dirs, just ignore the irrelevant `host`. I agree that's not completely correct. It could also copy everything but `host`, not make the decisions about services. But this is fine.